python/tests/table/test_sorting.py

```diff
@@ -0,0 +1,53 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from pyiceberg import transforms
+from pyiceberg.table.sorting import (
+    NullOrder,
+    SortDirection,
+    SortField,
+    SortOrder,
+)
+from pyiceberg.transforms import IdentityTransform
+from pyiceberg.types import IntegerType, StringType
+
+
+def test_serialize_sort_order():
```

Comment:
Does deserialization also work?

Reply:
I still need to implement that one, mostly because of the transform. `bucket[4]` cannot be deserialized to the implementation without knowing which type the source-id refers to.

Reply:
In Java, we recently introduced `UnboundSortOrder` and `UnboundPartitionSpec` to fix this problem. We may want to do the same here, where the transform is not instantiated until the order is bound to a schema.

Reply:
I was working on something similar, thanks!

Reply:
Just updated this with a `UnboundTransform` that deserializes properly back and forth, and can be bound using the `bind()` method. Also added a lot of tests 👍🏻